examples/signal/plot_covariance_estimation.py

```diff
@@ -0,0 +1,216 @@
+"""
+===============================================================================
+Estimate covariance with different time windows
+===============================================================================
+
+Covariance estimators comparison for different EEG signal length and
+impact on classification [1]_.
+"""
+# Author: Sylvain Chevallier
+#
+# License: BSD (3-clause)
+
+import numpy as np
+import pandas as pd
+from matplotlib import pyplot as plt
+from numpy.linalg import eigh
+from mne import Epochs, pick_types, events_from_annotations
+from mne.io import concatenate_raws
+from mne.io.edf import read_raw_edf
+from mne.datasets import eegbci
+from pyriemann.estimation import Covariances
+from pyriemann.utils.distance import distance
+from pyriemann.classification import MDM, TSclassifier
+
+from sklearn.model_selection import cross_val_score, StratifiedKFold
+from sklearn.pipeline import Pipeline
+from sklearn.linear_model import LogisticRegression
+from sklearn.pipeline import make_pipeline
+
+###############################################################################
+# Estimating covariance on synthetic data
+# ----------------------------------------
+#
+# Generate synthetic data, sampled from a distribution considered as the
+# groundtruth.
+
+Nt, Nc, Ns = 10, 5, 1000
+shape = (Nt, Nc, Nc)
+var = 2.0 + 0.1 * np.random.randn(Nt, Nc)
+A = 2 * np.random.rand(Nc, Nc) - 1
+A /= np.atleast_2d(np.sqrt(np.sum(A ** 2, 1))).T
+true_cov = np.empty(shape=shape)
+X = np.empty(shape=shape)
+for i in range(Nt):
+    true_cov[i] = A @ np.diag(var[i]) @ A.T
+    X[i] = np.random.multivariate_normal(np.array([0.0] * Nc), true_cov[i], size=Ns).T
+
+###############################################################################
+# Covariances() method offers several estimators: SCM, Ledoit-Wolf (LWF),
+# Schaefer-Strimmer (SCH), oracle approximating shrunk covariance (OAS),
+# minimum covariance determinant (MCD) and others. We will compare the
+# distance of LWF, OAS and SCH estimators with the groundtruth, while
+# increasing number of samples.
+
+estimators = ["lwf", "oas", "sch"]
+w_len = np.linspace(10, Ns, 100, dtype=np.int)
+dist_true = np.zeros(shape=(len(estimators), len(w_len), Nt))
+for i, e in enumerate(estimators):
+    for j, t in enumerate(w_len):
+        cov_est = Covariances(estimator=e).transform(X[:, :, :t])
+        for k in range(Nt):
+            dist_true[i, j, k] = distance(cov_est[k], true_cov[k], metric="riemann")
```

Comment:
this is very matlab style. I would suggest to use a DataFrame with 4 columns: estimator, n_times, n_trials, distance

then you can use simple pandas to plot things.